autopilot/migrator/migrator.go

```diff
@@ -0,0 +1,364 @@
+package migrator
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"net"
+	"sort"
+	"sync"
+	"time"
+
+	rhpv3 "go.sia.tech/core/rhp/v3"
+	"go.sia.tech/core/types"
+	"go.sia.tech/renterd/alerts"
+	"go.sia.tech/renterd/api"
+	"go.sia.tech/renterd/config"
+	"go.sia.tech/renterd/internal/accounts"
+	"go.sia.tech/renterd/internal/contracts"
+	"go.sia.tech/renterd/internal/download"
+	"go.sia.tech/renterd/internal/hosts"
+	"go.sia.tech/renterd/internal/memory"
+	"go.sia.tech/renterd/internal/rhp"
+	rhp4 "go.sia.tech/renterd/internal/rhp/v4"
+	"go.sia.tech/renterd/internal/upload"
+	"go.sia.tech/renterd/internal/utils"
+	"go.sia.tech/renterd/object"
+	"go.uber.org/zap"
+)
+
+const (
+	// migrationAlertRegisterInterval is the interval at which we update the
+	// ongoing migrations alert to indicate progress
+	migrationAlertRegisterInterval = 30 * time.Second
+
+	// migratorBatchSize is the amount of slabs we fetch for migration from the
+	// slab store at once
+	migratorBatchSize = math.MaxInt // TODO: change once we have a fix for the infinite loop
+)
+
+type (
+	Bus interface {
```

Comment:
This is horrible but figured it's probably best to tackle it when we move the autopilot into the bus.